src/OpenTelemetry/Internal/TagTransformer.cs

```diff
@@ -58,7 +60,7 @@ public bool TryTransformTag(KeyValuePair<string, object> tag, out T result, int?
                     // If an exception is thrown when calling ToString
                     // on any element of the array, then the entire array value
                     // is ignored.
-                    OpenTelemetrySdkEventSource.Log.UnsupportedAttributeType(tag.Value.GetType().ToString(), tag.Key);
+                    LogUnsupportedAttributeType?.Invoke(tag.Value.GetType().ToString(), tag.Key);
```

Comment:
Maybe put "ToString()" under the null check (so we can skip the `ToString()` call if the callback is null).

Reply:
We wouldn't be calling `ToString()` if the callback is `null` as we use the conditional operator `?` when calling `Invoke`: `LogUnsupportedAttributeType?.Invoke()`